Client-Server/Server/SeTCPServer.m

```diff
@@ -0,0 +1,88 @@
+#import "SeTCPServer.h"
+#import <Foundation/Foundation.h>
+#import <CoreFoundation/CoreFoundation.h>
+#import <sys/socket.h>
+#import <netinet/in.h>
+
+@interface SeTCPServer()
+
+@property (nonatomic) CFSocketRef socket;
+@property (nonatomic) NSInputStream *inputStream;
+
+@end
+
+@implementation SeTCPServer
+
+- (id)init {
+    return [self initWithPort:6666];
+}
+
+- (id)initWithPort:(NSUInteger)port {
+    if(([super init])) {
+        _port = port;
+    }
+
+    return self;
+}
+
+- (void)start {
+    self.socket = CFSocketCreate(kCFAllocatorDefault, PF_INET, SOCK_STREAM, IPPROTO_TCP, kCFSocketAcceptCallBack, serverConnectCallback, nil);
+   
+    if(!self.socket) {
+        NSLog(@"CFSocketCreate failed");
+        return;
+    }
+
+    struct sockaddr_in sin = {0};
+    sin.sin_len = sizeof(sin);
+    sin.sin_family = AF_INET; /* Address family */
+    sin.sin_port = htons(self.port); /* Or a specific port */
+    sin.sin_addr.s_addr = INADDR_ANY;
+
+    CFDataRef sinData = CFDataCreate(kCFAllocatorDefault, (UInt8 *)&sin, sizeof(sin));
+    if (CFSocketSetAddress(self.socket, sinData)) {
+        NSLog(@"CFSocketSetAddress failed");
+        return;
+    }
+
+    CFRunLoopSourceRef socketsource = CFSocketCreateRunLoopSource(kCFAllocatorDefault, self.socket, 0);
+    CFRunLoopAddSource(CFRunLoopGetCurrent(), socketsource, kCFRunLoopDefaultMode);
+    CFRunLoopRun();
```

Comment:
How'd you make the server stop? Add functionality that if the message "quit" is received from a client, the server stops. I'm commenting here because the source is needed for that :)